snap/squashfs/squashfs_test.go

```diff
@@ -962,3 +975,27 @@ func (s *SquashfsTestSuite) TestBuildWithCompressionUnhappy(c *C) {
 	})
 	c.Assert(err, ErrorMatches, "(?m)^mksquashfs call failed: ")
 }
+
+func (s *SquashfsTestSuite) TestBuildBelowMinimumSize(c *C) {
+	// this snap is empty. without truncating it to be larger, it should be smaller than
+	// the minimum snap size
+	sn := squashfs.New(filepath.Join(c.MkDir(), "truncate_me.snap"))
+	sn.Build(c.MkDir(), nil)
+
+	size, err := sn.Size()
+	c.Assert(err, IsNil)
+
+	c.Assert(int(size), testutil.IntEqual, int(squashfs.MinimumSnapSize))
+}
+
+func (s *SquashfsTestSuite) TestBuildAboveMinimumSize(c *C) {
+	// fill a snap with random data that will not compress well. it should be forced
+	// to be bigger than the minimum threshold
+	randomData := randutil.RandomString(int(squashfs.MinimumSnapSize * 2))
```

Comment:
I don't think this would be flaky, but I would appreciate other potential testing strategies if anyone has them.

Reply:
It would be nice if this were reproducible, such as by passing in a randomness seed, but then one might as well simply hard-code the test data. I think this is a good approach, out of convenience.